frontend/amundsen_application/static/js/ducks/lineage/reducer.ts

```diff
@@ -0,0 +1,227 @@
+import { ColumnLineageMap, Lineage } from 'interfaces/Lineage';
+import {
+  GetColumnLineage,
+  GetColumnLineageRequest,
+  GetColumnLineageResponse,
+  GetTableLineage,
+  GetTableLineageRequest,
+  GetTableLineageResponse,
+  GetTableColumnLineage,
+  GetTableColumnLineageRequest,
+  GetTableColumnLineageResponse,
+} from './types';
+
+export const initialLineageState = {
+  upstream_entities: [],
+  downstream_entities: [],
+  depth: 0,
+  direction: 'both',
+  key: '',
+};
+
+export const initialState: LineageReducerState = {
+  lineageTree: initialLineageState,
+  statusCode: null,
+  isLoading: false,
+  // ToDo: Please remove once list based view is deprecated
+  columnLineageMap: {},
+};
+
+/* ACTIONS */
+export function getTableLineage(
+  key: string,
+  depth: number = 1,
+  direction: string = 'both'
+): GetTableLineageRequest {
+  return {
+    type: GetTableLineage.REQUEST,
+    payload: { key, depth, direction },
+  };
+}
+
+export function getTableLineageSuccess(
+  data: Lineage,
+  statusCode: number
+): GetTableLineageResponse {
+  return {
+    type: GetTableLineage.SUCCESS,
+    payload: {
+      lineageTree: data,
+      statusCode,
+    },
+  };
+}
+
+export function getTableLineageFailure(
+  statusCode: number
+): GetTableLineageResponse {
+  return {
+    type: GetTableLineage.FAILURE,
+    payload: {
+      lineageTree: initialLineageState,
+      statusCode,
+    },
+  };
+}
+
+export function getColumnLineage(
+  key: string,
+  columnName: string,
+  depth: number = 1,
+  direction: string = 'both'
+): GetColumnLineageRequest {
+  return {
+    type: GetColumnLineage.REQUEST,
+    payload: { key, depth, direction, columnName },
+    meta: {
+      analytics: {
+        name: `getColumnLineage`,
```

Comment:
Let's namespace this one as `lineage/getColumnLineage`:
```suggestion
        name: `lineage/getColumnLineage`,
```